homeassistant/helpers/area_registry.py

```diff
@@ -45,27 +46,47 @@ def __init__(self, hass: HomeAssistantType) -> None:
         self.hass = hass
         self.areas: MutableMapping[str, AreaEntry] = {}
         self._store = hass.helpers.storage.Store(STORAGE_VERSION, STORAGE_KEY)
+        self._normalized_name_area_idx: Dict[str, str] = {}
 
     @callback
     def async_get_area(self, area_id: str) -> Optional[AreaEntry]:
-        """Get all areas."""
+        """Get area by id."""
         return self.areas.get(area_id)
 
+    @callback
+    def async_get_area_by_name(self, name: str) -> Optional[AreaEntry]:
+        """Get area by name."""
+        normalized_name = normalize_area_name(name)
+        if normalized_name not in self._normalized_name_area_idx:
+            return None
+        return self.areas.get(self._normalized_name_area_idx[normalized_name])
```

Comment:
Use square brackets. If the index is out of date, it's better to crash so we can fix it.

```suggestion
        return self.areas[self._normalized_name_area_idx[normalized_name]]
```

Reply:
If it happens we have a bug. 👍